### PR TITLE
10740: elims mega message

### PR DESCRIPTION
**Pull request by mkalish**
This PR adds new translations for ELIMS, specifically handling specimen

Test Steps:
Using both sample messages in the linked issue
1. Update the `dev.DEV_ELIMS` receiver to be active
2. Send both messages from the `ignore.ignore-elr-elims` sender
3. Verify that the only discrepancies between the two are noted in the spreadsheets
https://cdc.sharepoint.com/:x:/r/teams/ReportStream/_layouts/15/Doc.aspx?sourcedoc=%7B2F962BDE-BCA1-4C16-B99E-7E8D31F3635D%7D&file=47_04608646_11024_MegaSpecimen.xlsx&action=default&mobileredirect=true
https://cdc.sharepoint.com/:x:/r/teams/ReportStream/_layouts/15/Doc.aspx?sourcedoc=%7B16FE9888-8743-466C-B78B-7F60D621129F%7D&file=47_32361_04608646_11034_MegaCase.xlsx&action=default&mobileredirect=true

## Changes
- Adds new HL7 -> FHIR extensions for specimen
- Updates the specimen translate schema to include all specimen values in the ELIMS message

### Testing
- [x] Tested locally?
- [x] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [x] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- #10740

### CODE REVIEW

File: prime-router/build.gradle.kts [hunk 1]
```diff
@@ -583,16 +583,24 @@ task<Exec>("uploadSwaggerUI") {
     commandLine("./upload_swaggerui.sh")
 }
 
+tasks.register("killFunc") {
+    exec {
+        workingDir = project.rootDir
```
Comment (mkalish) [inline prime-router/build.gradle.kts:588]: Not related, but this is a crude fix for the the issue that azure func no longer responds to `SIGINT`

File: prime-router/src/main/kotlin/azure/SettingsFunctions.kt
Comment (mkalish) [inline prime-router/src/main/kotlin/azure/SettingsFunctions.kt:1]: Not sure how this snuck in, but there were several linting errors in master

File: prime-router/metadata/fhir_mapping/hl7/resource/Specimen.yml [hunk 2]
```diff
@@ -29,6 +29,23 @@ type:
   expressionType: resource
   specs: SPM.4 | SPM.5
 
+extension_1-specimenTypeModifier:
+  condition: $spm5 NOT_NULL && $spm4 NOT_NULL
```
Comment (mkalish) [inline prime-router/metadata/fhir_mapping/hl7/resource/Specimen.yml:33]: I wasn't 100% what 30L is guarding against, but it seemed like a fair check regardless to only map this when both segments are populated

File: prime-router/metadata/fhir_mapping/hl7/resource/Specimen.yml [hunk 3]
```diff
@@ -63,10 +80,11 @@ note_1:
     noteText: SPM.14
 
 container:
-  condition: $containerTypeCWE NOT_NULL
+  condition: $containerTypeCWE NOT_NULL || $specimenAdditive NOT_NULL
```
Comment (mkalish) [inline prime-router/metadata/fhir_mapping/hl7/resource/Specimen.yml:83]: The ELIMs messages had SPM.6 but not SPM.27

**Review by JFU-NAVA-PBC — COMMENTED** [review]
suggest drive by fix DEBUG logging template:

Verb ending with 'e' result in bellow logging:

DEBUG gov.cdc.prime.router.fhirengine.azure.FHIRFunctions - **Translateing** message: 
DEBUG gov.cdc.prime.router.fhirengine.azure.FHIRFunctions - **Routeing** 

// logger.debug("${engineType}ing message: $message for the $dequeueCount time")
val verb = if engineType.endsWith("e") engineType.substring(0, engineType.length - 1) else engineType 
logger.debug("${verb}ing message: $message for the $dequeueCount time")

**Review by JFU-NAVA-PBC — COMMENTED** [review]
MSH.18:

ELIMS: MSH|^\~\&#|STARLIMS.CDC.Stag^2.16.840.1.114222.4.3.3.2.1.2^ISO|CDC Atlanta^11D0668319^CLIA|CDPH CA CALREDIE^2.16.840.1.114222.4.3.3.10.1.1^ISO|CDPH_CID^2.16.840.1.114222.4.1.2.14104^ISO|20230213111654-0500||ORU^R01^ORU_R01|32361_U47 Mega Case Take 1_3015894676_04608646_11034|T|2.5.1|||NE|NE|USA|**=============**|||PHLabReport-NoAck^PHIN^2.16.840.1.113883.9.11^ISO
RSUP:  MSH|^\~\&#|STARLIMS.CDC.Stag^2.16.840.1.114222.4.3.3.2.1.2^ISO|CDC Atlanta^11D0668319^CLIA|CDPH CA CALREDIE^2.16.840.1.114222.4.3.3.10.1.1^ISO|CDPH_CID^2.16.840.1.114222.4.1.2.14104^ISO|20230213111654-0500||ORU^R01^ORU_R01|32361_U47 Mega Case Take 1_3015894676_04608646_11034|T|2.5.1|||NE|NE|USA|**UNICODE UTF-8**|||PHLabReport-NoAck^PHIN^2.16.840.1.113883.9.11^ISO

Question: MSH.18 Character Set is Optional, RSUP translate put a value: UNICODE UTF-8 in there, this is already commented in the spread sheet

however, if change MSH.18 to 8891/1 and submitted the report, as shown below, the char encoding in original message has not been translated into the final hl7 message at receiver end, instead it is still the :

ELIMS: MSH|^~\&#|STARLIMS.CDC.Stag^2.16.840.1.114222.4.3.3.2.1.2^ISO|CDC Atlanta^11D0668319^CLIA|CDPH CA CALREDIE^2.16.840.1.114222.4.3.3.10.1.1^ISO|CDPH_CID^2.16.840.1.114222.4.1.2.14104^ISO|20230213104839-0500||ORU^R01^ORU_R01|3015894676_04608646_11024|T|2.5.1|||NE|NE|USA|**8859/1**|||PHLabReport-NoAck^PHIN^2.16.840.1.113883.9.11^ISO
RSUP:  MSH|^~\&#|STARLIMS.CDC.Stag^2.16.840.1.114222.4.3.3.2.1.2^ISO|CDC Atlanta^11D0668319^CLIA|CDPH CA CALREDIE^2.16.840.1.114222.4.3.3.10.1.1^ISO|CDPH_CID^2.16.840.1.114222.4.1.2.14104^ISO|20230213104839-0500||ORU^R01^ORU_R01|3015894676_04608646_11024|T|2.5.1|||NE|NE|USA|**UNICODE UTF-8**|||PHLabReport-NoAck^PHIN^2.16.840.1.113883.9.11^ISO

**Review by JFU-NAVA-PBC — COMMENTED** [review]
NTE.4: 

Note, '=====' used as place holder for missing value.

Original ELIMS message have extra field per HL7 v2 2.5.1:
NTE|1|L|test summary   two lines|RE^Remark^HL70364^^^^2.5.1^^^^^^^2.16.840.1.113883.12.364
NTE|1|L|test summary   two lines|RE^Remark^HL70364========================================

Field 3: L	Ancillary (filler) department is source of comment

Field 4: CWE

OID: 2.16.840.1.113883.12.364	364	0	0	Health Level 7 (HL7) version 2.x Comment type used in chapter(s) 2
Comment type

May be the sending application is on HL7 v2 2.7?? OR the sending application generated invalid message??

**Review by JFU-NAVA-PBC — COMMENTED** [review]
Same ask about OCR, OBR, OBX as Micheal commented in spread sheet, e..g timestamp format: 
Note: "=================" used in message as place holder for missing value...

OBR.4:

**ELIMS:** 
OBR|1|sphlspid^SPHL-000048^2.16.840.1.114222.4.1.10765^ISO|3015894676_04608646^STARLIMS.CDC.Stag^2.16.840.1.114222.4.3.3.2.1.2^ISO|68991-9^Epidemiologically Important Information^LN^^^^2.69^^^CDC-10516^Poxvirus Serology^L^^2.16.840.1.113883.6.1|||202302101135=======|||||||||SPHL-000148^CA-Veterans Affairs Palo Alto Healthcare System^^^^^^^STARLIMS.CDC.Stag&2.16.840.1.114222.4.3.3.2.1.2&ISO^^^^XX|^NET^Internet^mark.holodniy@va.gov|||||202302131116==-0500|||F
**RSUP:**  
OBR|1|sphlspid^SPHL-000048^2.16.840.1.114222.4.1.10765^ISO|3015894676_04608646^STARLIMS.CDC.Stag^2.16.840.1.114222.4.3.3.2.1.2^ISO|68991-9^Epidemiologically Important Information^LN===============================================================|||20230210113500+0000|||||||||SPHL-000148^CA-Veterans Affairs Palo Alto Healthcare System^^^^^^^STARLIMS.CDC.Stag&2.16.840.1.114222.4.3.3.2.1.2&ISO^^^^XX|^NET^Internet^mark.holodniy@va.gov|||||20230213111600-0500|||F

OBX.3:
OBX.6:

**ELIMS:** 
OBX|1|CWE|100888-7^West African MVPX DNA Spec Ql NAA+probe^LN^3905^Monkeypox WA specific^L^2.69^v_unknown^Monkeypox WA specific|ZZYGNASR-1|10828004^Positive^SCT^^^^09012018^^Positive||||||C|||202302101135=======|11D0668319^Centers for Disease Control and Prevention^CLIA^47^Poxvirus Laboratory/Poxvirus and Rabies Branch^L|UIE8@CDC.GOV^Perkins ^Kayla|||20230213101829=====||||Centers for Disease Control and Prevention^L^^^^CLIA&2.16.840.1.113883.4.7&ISO^XX^^^11D0668319|1600 Clifton Rd^^Atlanta^GA^30329^USA^B
**RSUP:**  
OBX|1|CWE|100888-7^West African MVPX DNA Spec Ql NAA+probe^LN^3905^Monkeypox WA specific^L=====================================|ZZYGNASR-1|10828004^Positive^SCT^^^^09012018^^Positive||||||C|||20230210113500+0000|11D0668319^Centers for Disease Control and Prevention^CLIA^47^Poxvirus Laboratory/Poxvirus and Rabies Branch^L|UIE8@CDC.GOV^Perkins ^Kayla|||20230213101829+0000||||Centers for Disease Control and Prevention^L^^^^CLIA&2.16.840.1.113883.4.7&ISO^XX^^^11D0668319|1600 Clifton Rd^^Atlanta^GA^30329^USA^B

**ELIMS:** 
OBX|1|CWE|68993-5^Human RNase P RNA XXX Ql NAA+probe^LN^3844^RNaseP human DNA^L^2.69^v_unknown^RNaseP human DNA|ZZYGNASR-1|260385009^Negative^SCT^^^^09012018^^Negative (No human DNA present)||||||C|||202302101135|11D0668319^Centers for Disease Control and Prevention^CLIA^47^Poxvirus Laboratory/Poxvirus and Rabies Branch^L|UIE8@CDC.GOV^Perkins ^Kayla|||20230213102132||||Centers for Disease Control and Prevention^L^^^^CLIA&2.16.840.1.113883.4.7&ISO^XX^^^11D0668319|1600 Clifton Rd^^Atlanta^GA^30329^USA^B
**RSUP:**  
OBX|1|CWE|68993-5^Human RNase P RNA XXX Ql NAA+probe^LN^3844^RNaseP human DNA^L================================|ZZYGNASR-1|260385009^Negative^SCT^^^^09012018^^Negative (No human DNA present)||||||C|||20230210113500+0000|11D0668319^Centers for Disease Control and Prevention^CLIA^47^Poxvirus Laboratory/Poxvirus and Rabies Branch^L|UIE8@CDC.GOV^Perkins ^Kayla|||20230213102132+0000||||Centers for Disease Control and Prevention^L^^^^CLIA&2.16.840.1.113883.4.7&ISO^XX^^^11D0668319|1600 Clifton Rd^^Atlanta^GA^30329^USA^B

**Comment by JFU-NAVA-PBC** [comment]
@mkalish Feel free to bring in more pair of eyes on this, thanks!

File: prime-router/metadata/hl7_mapping/ORU_R01/base/specimen.yml [hunk 4]
```diff
@@ -47,16 +47,25 @@ elements:
     schema: ../../common/datatype/ce-coded-with-exceptions
 
   - name: specimen-type-modifier
-    value: [ '""' ]
-    hl7Spec: [ '%{hl7SpecimenFieldPath}-5' ]
+    resource: '%resource.extension("https://reportstream.cdc.gov/fhir/StructureDefinition/specimen-type-modifier").value'
```
Comment (victor-chaparro) [inline prime-router/metadata/hl7_mapping/ORU_R01/base/specimen.yml:50]: When getting extensions, you can re-use the constant for the RS extension URL defined at the top of `ORU_R01-base.yml` and get the value this way `.extension(%`rsext-specimen-type-modifier`).value`

**Review by arnejduranovic — APPROVED** [review]

**Pull request merged**